blosc/schunk.c

```diff
@@ -141,19 +143,46 @@ blosc2_schunk* blosc2_schunk_new(const blosc2_storage storage) {
   // ...and update internal properties
   update_schunk_properties(schunk);
 
-  if (storage.sequential) {
+  if (!storage.sequential && storage.path != NULL){
+    char* urlpath;
+    char last_char = storage.path[strlen(storage.path) - 1];
+    if (last_char == '\\' || last_char == '/') {
+      urlpath = malloc(strlen(storage.path));
+      strncpy(urlpath, storage.path, strlen(storage.path) - 1);
+      urlpath[strlen(storage.path) - 1] = '\0';
+    }
+    else {
+      urlpath = malloc(strlen(storage.path) + 1);
+      strcpy(urlpath, storage.path);
+    }
+    //Create directory
+    if (mkdir(urlpath,0777) == -1) {
+      BLOSC_TRACE_ERROR("Error during the creation of the directory, maybe it already exists.");
+      return NULL;
+    }
     // We want a frame as storage
-    blosc2_frame* frame = blosc2_frame_new(storage.path);
+    blosc2_frame* frame = blosc2_frame_new(urlpath);
+    free(urlpath);
+    frame->eframe = true;
     // Initialize frame (basically, encode the header)
     int64_t frame_len = blosc2_frame_from_schunk(schunk, frame);
     if (frame_len < 0) {
       BLOSC_TRACE_ERROR("Error during the conversion of schunk to frame.");
+      return NULL;
     }
     schunk->frame = frame;
   }
-  else if (storage.path != NULL) {
-    BLOSC_TRACE_ERROR("Creating empty sparse schunks on-disk is not supported yet.");
-    return NULL;
+  if (storage.sequential){
+    // We want a frame as storage
+    blosc2_frame* frame = blosc2_frame_new(storage.path);
```

Comment:
We probably want to rename `storage.path` -> `storage.urlpath`

Reply:
Fixed in 9843e19b897bccaf7952b5bd7f988e89fadac8fb